### DISCUSSION TITLE
3.0.3 has a problem with the maven repository. The size is too large.

### DISCUSSION THREAD

**Discussion by ouyang-xtt** [apache/pulsar#22296] [Q&A]
https://repo1.maven.org/maven2/org/apache/pulsar/pulsar/3.0.3/
![image](https://github.com/apache/pulsar/assets/96647256/162d804a-661b-4bc6-ae7e-a5f76e7284ab)

An error is reported during download.
![image](https://github.com/apache/pulsar/assets/96647256/bc53d9c8-84cd-4587-acc2-d377b77fa518)

**Comment by lhotari** [comment]
It seems that we shouldn't upload the .tar.gz to maven at all. Please find the correct download link at https://pulsar.apache.org/download/ .

**Reply by visortelle** [reply]
FYI,
I don't know the specific author's use case, but it's easy to be permanently banned by `archive.apache.org`:

> If you break one of these rules, your IP will be banned from all services across the ASF.
> More than 40GB downloaded from archive.apache.org per week.

https://infra.apache.org/infra-ban.html

**Reply by visortelle** [reply]
The same issue for [pulsar-3.0.3-source-release.zip](https://repo1.maven.org/maven2/org/apache/pulsar/pulsar/3.0.3/pulsar-3.0.3-source-release.zip)

but works normally for [pulsar-3.2.1-source-release.zip](https://repo1.maven.org/maven2/org/apache/pulsar/pulsar/3.2.1/pulsar-3.2.1-source-release.zip)

also works normally for 3.0.2 [pulsar-3.0.2-source-release.zip](https://repo1.maven.org/maven2/org/apache/pulsar/pulsar/3.0.2/pulsar-3.0.2-source-release.zip)

The file size is significantly bigger specifically for the 3.0.3

<img width="663" alt="Screenshot 2024-03-19 at 11 47 23 AM" src="https://github.com/apache/pulsar/assets/9302460/01b91a3b-6f65-49cc-82ab-fe1a59387b6d">
<img width="645" alt="Screenshot 2024-03-19 at 11 47 30 AM" src="https://github.com/apache/pulsar/assets/9302460/75aaf1fb-6ae2-46c9-b159-9dd11a14780e">

**Reply by lhotari** [reply]
@visortelle this isn't about archive.apache.org. The report contained the url https://repo1.maven.org/maven2/org/apache/pulsar/pulsar/3.0.3/ .

**Reply by visortelle** [reply]
I understand, but you recommended using the links on the https://pulsar.apache.org/download/ page.

This page contains links to `archive.apache.org`. 

I was banned once because I didn't know this limit and wanted to warn the author just in case.

For example, you can't rely on these links in case you use self-hosted CI runners with a fixed IP.

<img width="619" alt="Screenshot 2024-03-19 at 12 26 25 PM" src="https://github.com/apache/pulsar/assets/9302460/29efa7d5-9fab-4c33-8cc5-f1df6bb32873">

As you can see in my second comment, something wrong with file sizes specifically for `3.0.3`

**Reply by visortelle** [reply]
@lhotari the https://archive.apache.org/ states:

> If you are looking for current software releases, please visit one of our numerous [mirrors](http://www.apache.org/dyn/closer.cgi).

Currently, only the current (3.2.1) Pulsar version uses mirrors on the downloads page.
LTS version (3.0.3) should also use the mirror instead of `archive.apache.org`.

**Reply by visortelle** [reply]
BTW probably, it would be useful to have these tables with links for LTS too. Probably as tabs to save vertical space.

```
[Current 3.2.1 | LTS 3.0.3]
----------------
Shell
| ------------- |
Offloaders
| ------------- |
Connectors
| ------------- |
...
```

<img width="1364" alt="Screenshot 2024-03-19 at 12 54 17 PM" src="https://github.com/apache/pulsar/assets/9302460/e1458f06-9618-4fba-ac21-e03472f130a3">

**Reply by ouyang-xtt** [reply]
Our team needs to select the Pulsar. After the selection, the tool downloads the source code from https://repo1.maven.org/maven2/ to our repository.During this process, the download fails because the size of the source code is abnormal.I don't know who uploaded this to the maven repository.Is this a pulsar problem or a maven problem?

**Reply by lhotari** [reply]
> As you can see in my second comment, something wrong with file sizes specifically for 3.0.3

@visortelle @ouyang-xtt sorry, I missed this detail. It seems that something went wrong during the release process of 3.0.3. I think we might need to release 3.0.4 to fix it. /Cc @heesung-sn

**Reply by heesung-sohn** [reply]
Oh, interesting; I believe I followed this process: https://pulsar.apache.org/contribute/release-process/#stage-maven-modules. It seems like something went wrong in the middle.

**Reply by heesung-sohn** [reply]
Can Pulsar PMC delete this maven dir? I can try to push this again after that.
